cpp/tests/prims/mg_update_frontier_v_push_if_out_nbr.cu

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/high_res_clock.h>
+#include <utilities/base_fixture.hpp>
+#include <utilities/device_comm_wrapper.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/test_utilities.hpp>
+#include <utilities/thrust_wrapper.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/partition_manager.hpp>
+#include <cugraph/utilities/dataframe_buffer.cuh>
+
+#include <cuco/detail/hash_functions.cuh>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/matrix_partition_view.hpp>
+#include <cugraph/prims/copy_to_adj_matrix_row_col.cuh>
+#include <cugraph/prims/row_col_properties.cuh>
+#include <cugraph/prims/update_frontier_v_push_if_out_nbr.cuh>
+#include <cugraph/prims/vertex_frontier.cuh>
+
+#include <thrust/copy.h>
+#include <thrust/count.h>
+#include <thrust/equal.h>
+#include <thrust/iterator/permutation_iterator.h>
+#include <raft/comms/comms.hpp>
+#include <raft/comms/mpi_comms.hpp>
+#include <raft/handle.hpp>
+#include <rmm/device_scalar.hpp>
+#include <rmm/device_uvector.hpp>
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <random>
+
+template <typename TupleType, typename T, std::size_t... Is>
+__device__ __host__ auto make_type_casted_tuple_from_scalar(T val, std::index_sequence<Is...>)
+{
+  return thrust::make_tuple(
+    static_cast<typename thrust::tuple_element<Is, TupleType>::type>(val)...);
+}
+
+template <typename property_t, typename T>
+__device__ __host__ auto make_property_value(T val)
```

Comment:
I assume this (make_property_value & property_transform_t) is a refactoring of https://github.com/rapidsai/cugraph/blob/branch-22.02/cpp/tests/prims/mg_extract_if_e.cu#L53, and could you make the same change in https://github.com/rapidsai/cugraph/blob/branch-22.02/cpp/tests/prims/mg_extract_if_e.cu#L53 as well?

We later need to make all the prims tests as similar as possible anyways (so once a user understands one example, the user can pick up remaining examples very quickly).